src/e2e-test/java/io/cdap/plugin/salesforcestreamingsource/stepsdesign/DesignTimeSteps.java

```diff
@@ -66,4 +67,54 @@ public void createNewLeadInSalesforce() throws UnsupportedEncodingException {
   public void fillUniqueTopicNameInRuntimeArguments(String runtimeArgumentKey) {
     SalesforcePropertiesPageActions.fillUniqueTopicNameInRuntimeArguments(runtimeArgumentKey);
   }
+
+  @Then("Update existing salesforce records")
+  public void updateExistingSalesforceRecords() {
+    String uniqueRecordId = SalesforceClient.queryObjectId(customObject);
+    SalesforceClient.updateObject(uniqueRecordId, customObject);
+  }
+
+
+  @And("Create a new Custom Record in salesforce using REST API")
+  public void createANewRecord() throws UnsupportedEncodingException,
+    InterruptedException {
+    // Waiting to create data in runtime
+    TimeUnit time = TimeUnit.SECONDS;
+    time.sleep(120);
```

Comment:
We are using this because without applying wait before creating a custom record the records will not getting transferred.
So that's why we have to use wait in runtime before creating a record.

Reply:
I understand why you would want to wait *after* creating the record in Salesforce (it takes some time for the streaming pipeline to process the newly created record). However, I don't understand why you need to wait for 120 seconds *before* you create the record in Salesforce (i.e. seems like the sleep should be after ` SalesforceClient.createObject(custom, customObject);`).

Can you please clarify?

Reply:
This wait is added here because when the pipeline moves to running status and I create the record in salesforce instantly that record take too much time to reflect on the plugin/logs and on the other hand if we wait a bit before creating the record in Salesforce it reflects much quickly. That's why added the wait here.